### PR TITLE
[NFC][LoopVectorize] Add more simple early exit tests

### PR DESCRIPTION
**Pull request by david-arm**
I realised we are missing tests to cover more loops with multiple early exits - some countable and some uncountable.

I've also added a few SVE versions of the test in the AArch64 directory. Once we can vectorise such early exit loops it's a good sanity check to make sure they also vectorise for SVE. Also, for some of the tests I expect there to be some divergence from the same tests in the top level directory once we start vectorising them.

### CODE REVIEW

**Comment by david-arm** [comment]
Since this patch is just adding tests, if there are no objections I'll merge the patch tomorrow morning!

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

File: llvm/test/Transforms/LoopVectorize/simple_early_exit.ll [hunk 1]
```diff
@@ -918,6 +917,189 @@ loop.end:
 }
 
 
+define i64 @multiple_exiting_one_early_same_exit_phi_of_consts() {
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/simple_early_exit.ll:920]: This file is getting a bit big , perhaps worth splitting it up into multiple files if possible?

Comment (david-arm) [reply llvm/test/Transforms/LoopVectorize/simple_early_exit.ll:920]: Sure, that makes sense. If it's ok with you, I'll land this patch first then follow up with a simple refactor that collects tests of the same class in their own file, i.e. early exits with no outside users of loop variables, ones that use loop variables, tests with multiple early exits (countable + uncountable), tests for different memory access patterns, etc.

**Pull request merged**